[Merged by Bors] - Added docs for bevy_transform

Sheepyhead opened the pull request:
# Objective

bevy_transform needed documentation and warn(missing_docs) as requested by #3492 

## Solution

warn(missing_docs) was activated and documentation was added to cover the crate

KDecay commented on crates/bevy_transform/src/components/parent.rs:
```diff
@@ -7,6 +7,7 @@ use bevy_ecs::{
 use bevy_reflect::Reflect;
 use std::ops::{Deref, DerefMut};
 
+/// Holds a reference to the parent entity of this entity, should only be present on entities that actually have a parent entity
```
I would split this up into two sentences for better readability.

```suggestion
/// Holds a reference to the parent entity of this entity.
/// This component should only be present on entities that actually have a parent entity.
```

KDecay commented on crates/bevy_transform/src/hierarchy/child_builder.rs:
```diff
@@ -123,39 +129,50 @@ impl Command for RemoveChildren {
         e
     }
 
+    /// Spawns an entity with no components and inserts in into the children defined by the ChildBuilder (adding the Parent component to it)
```
```suggestion
    /// Spawns an [`Entity`] with no components and inserts it into the children defined by the ChildBuilder which adds the [`Parent`] component to it.
```

KDecay commented on crates/bevy_transform/src/hierarchy/child_builder.rs:
```diff
@@ -240,6 +259,7 @@ impl<'w> WorldChildBuilder<'w> {
         self.world.entity_mut(entity)
     }
 
+    /// Spawns an entity with no components and inserts in into the children defined by the WorldChildBuilder (adding the Parent component to it)
```
```suggestion
    /// Spawns an [`Entity`] with no components and inserts it into the children defined by the WorldChildBuilder which adds the [`Parent`] component to it.
```

KDecay reviewed (changes requested):
The documentation looks great. I only found some smaller things that need to be fixed. 

One thing I noticed is that you didn't highlight/link to the documented types using [``]. I added this in two suggestions I made, but it would be great to have them used as much as possible in my opinion. It just helps the user navigate on docs.rs.

mockersf commented on crates/bevy_transform/src/components/global_transform.rs:
```diff
@@ -36,8 +36,11 @@ use std::ops::Mul;
 #[derive(Component, Debug, PartialEq, Clone, Copy, Reflect)]
 #[reflect(Component, PartialEq)]
 pub struct GlobalTransform {
+    /// The position of the transform
```
```suggestion
    /// The position of the global transform
```
I would prefer to avoid "transform" without global in this file (unless the doc is speaking about a transform) to reduce confusion between transforms and global transforms

mockersf commented:
The doc comments could use a few more links, and some of the changes here would break with https://github.com/bevyengine/bevy/pull/3457 without backquotes

Sheepyhead commented:
> The documentation looks great. I only found some smaller things that need to be fixed.
> 
> One thing I noticed is that you didn't highlight/link to the documented types using [``]. I added this in two suggestions I made, but it would be great to have them used as much as possible in my opinion. It just helps the user navigate on docs.rs.



> The doc comments could use a few more links, and some of the changes here would break with #3457 without backquotes

I updated linking in the docs, I might have missed some places

KDecay reviewed (commented):
Does the following command run through fine? I know that you don't like that we are adding the `clippy::doc_markdown`lint (see #3492), but it would be nice if we don't create any issues when we activate it later on:

`cargo clippy -p bevy_transform --no-deps --all-targets --all-features -- -D warnings -A clippy::type_complexity -W clippy::doc_markdown`

Sheepyhead commented:
> Does the following command run through fine? I know that you don't like that we are adding the `clippy::doc_markdown`lint (see #3492), but it would be nice if we don't create any issues when we activate it later on:
> 
> `cargo clippy -p bevy_transform --no-deps --all-targets --all-features -- -D warnings -A clippy::type_complexity -W clippy::doc_markdown`

I did check it against that yes

DJMcNab commented on crates/bevy_transform/src/components/children.rs:
```diff
@@ -7,6 +7,7 @@ use bevy_reflect::Reflect;
 use smallvec::SmallVec;
 use std::ops::Deref;
 
+/// Contains references to the child entities of this entity
```
To me, references makes me think of `&T`.

Possibly references is the right term; if it is, we should ensure we're consistent about it (but that's not this PR's problem)

DJMcNab reviewed (approved):
I think having this in is better than not merging it, although I can't help feeling that some of the 'sub' docs are slightly superfluous; e.g. documenting all fields of the commands.

Why does one of the commands have pub fields and the others don't?

mockersf commented:
> Why does one of the commands have pub fields and the others don't?

`AddChild` was added recently, with the goal of being able to be executed without going through a `Commands` buffer. Fields needed to be pub as it was to run it from another Bevy crate. I asked if there was a reason to keep the fields not pub but no answer 🙂 

I think most commands should have pub fields, as it's possible to add commands directly but most are not possible to build... but that's for another PR

alice-i-cecile commented:
bors r+
